Exception Details on  Build step io.quarkus.arc.deployment.ArcAnnotationProcessor#build

**Description**
When there are a injected bean in a abstract class that is extended for the implementation of the same bean, we have this Exception on start quarkus dev:
`ERROR [io.qua.dev.DevModeMain] Failed to start quarkus: java.lang.RuntimeException: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
	[error]: Build step io.quarkus.arc.deployment.ArcAnnotationProcessor#build threw an exception: java.lang.NullPointerException
	at io.quarkus.runner.RuntimeRunner.run(RuntimeRunner.java:137)
	at io.quarkus.dev.DevModeMain.doStart(DevModeMain.java:159)
	at io.quarkus.dev.DevModeMain.main(DevModeMain.java:93)
Caused by: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
	[error]: Build step io.quarkus.arc.deployment.ArcAnnotationProcessor#build threw an exception: java.lang.NullPointerException
	at io.quarkus.builder.Execution.run(Execution.java:124)
	at io.quarkus.builder.BuildExecutionBuilder.execute(BuildExecutionBuilder.java:137)
	at io.quarkus.deployment.QuarkusAugmentor.run(QuarkusAugmentor.java:108)
	at io.quarkus.runner.RuntimeRunner.run(RuntimeRunner.java:102)
	... 2 more
Caused by: java.lang.NullPointerException
	at io.quarkus.gizmo.BytecodeCreatorImpl.allocateLocalVariables(BytecodeCreatorImpl.java:906)
	at io.quarkus.gizmo.MethodCreatorImpl.write(MethodCreatorImpl.java:104)
	at io.quarkus.gizmo.ClassCreator.close(ClassCreator.java:172)
	at io.quarkus.arc.processor.ComponentsProviderGenerator.generate(ComponentsProviderGenerator.java:192)
	at io.quarkus.arc.processor.BeanProcessor.process(BeanProcessor.java:217)
	at io.quarkus.arc.deployment.ArcAnnotationProcessor.build(ArcAnnotationProcessor.java:265)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:483)
	at io.quarkus.deployment.ExtensionLoader$1.execute(ExtensionLoader.java:507)
	at io.quarkus.builder.BuildContext.run(BuildContext.java:414)
	at io.quarkus.builder.BuildContext$$Lambda$71/148735482.run(Unknown Source)
	at org.jboss.threads.ContextClassLoaderSavingRunnable.run(ContextClassLoaderSavingRunnable.java:35)
	at org.jboss.threads.EnhancedQueueExecutor.safeRun(EnhancedQueueExecutor.java:2011)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.doRunTask(EnhancedQueueExecutor.java:1538)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1429)
	at java.lang.Thread.run(Thread.java:745)
	at org.jboss.threads.JBossThread.run(JBossThread.java:479)`
This is a generic exception.

**How to reproduce**
The abstract Class
`public abstract class AbstractServiceImpl {
	@Inject
	protected AppMessages appMessages;
 }
 }` 
The implementation class
`@RequestScoped
 public class AppMessagesImpl extends AbstractServiceImpl implements AppMessages {
	@Inject
	AppMessagesLoader messagesLoader;
 }`


**Implementation ideas**
My suggestion is throws an specific Exception showing more details about the error.

--- a/independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/ComponentsProviderGenerator.java
+++ b/independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/ComponentsProviderGenerator.java
@@ -272,9 +272,7 @@ public class ComponentsProviderGenerator extends AbstractGenerator {
     private boolean isDependency(BeanInfo bean, Map<BeanInfo, List<BeanInfo>> beanToInjections) {
         for (Iterator<Entry<BeanInfo, List<BeanInfo>>> iterator = beanToInjections.entrySet().iterator(); iterator.hasNext();) {
             Entry<BeanInfo, List<BeanInfo>> entry = iterator.next();
-            if (entry.getKey().equals(bean)) {
-                continue;
-            } else if (entry.getValue().contains(bean)) {
+            if (entry.getValue().contains(bean)) {
                 return true;
             }
         }

--- a/independent-projects/arc/tests/src/test/java/io/quarkus/arc/test/injection/erroneous/CircularInjectionNotSupportedTest.java
+++ b/independent-projects/arc/tests/src/test/java/io/quarkus/arc/test/injection/erroneous/CircularInjectionNotSupportedTest.java
@@ -1,0 +1,54 @@
+package io.quarkus.arc.test.injection.erroneous;
+
+import io.quarkus.arc.test.ArcTestContainer;
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.RuleChain;
+import org.junit.rules.TestRule;
+import org.junit.runner.Description;
+import org.junit.runners.model.Statement;
+
+public class CircularInjectionNotSupportedTest {
+
+    @Rule
+    public RuleChain chain = RuleChain.outerRule(new TestRule() {
+        @Override
+        public Statement apply(Statement base, Description description) {
+            return new Statement() {
+                @Override
+                public void evaluate() throws Throwable {
+                    try {
+                        base.evaluate();
+                    } catch (IllegalStateException e) {
+                        // expected failure on ISE due to circular dependency
+                    }
+                }
+            };
+        }
+    }).around(new ArcTestContainer(Foo.class, AbstractServiceImpl.class, ActualServiceImpl.class));
+
+    @Test
+    public void testExceptionThrown() {
+        // throws exception during deployment
+    }
+
+    static abstract class AbstractServiceImpl {
+        @Inject
+        protected Foo foo;
+    }
+
+    @ApplicationScoped
+    static class ActualServiceImpl extends AbstractServiceImpl implements Foo {
+
+        @Override
+        public void ping() {
+        }
+    }
+
+    interface Foo {
+        void ping();
+    }
+
+}

